### PR TITLE
Corrected analyzer for SA1623 and SA1624 to report the correct diagnostic

### LINKED ISSUE #2253: SA1623 - Incorrect Error on Read-Only Peoperty
Consider a read only property with read/write verbiage copypasta:
```
/// <summary>  
/// Gets or sets the path to the object's class.  
/// </summary>  
public ManagementPath ClassPath => this.managementObject.ClassPath;
```

SA1623 tags this with the misleading message "text must begin with: Gets", and auto-fix changes the summary to "Gets gets or sets".

### PR DESCRIPTION
**Pull request by bjornhellander**
Fixes #2253. Also found a number of similar cases that were not reported correctly either.

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/PropertySummaryDocumentationAnalyzer.cs [hunk 1]
```diff
@@ -130,7 +130,10 @@ private static void AnalyzeSummaryElement(SyntaxNodeAnalysisContext context, Xml
             var textElement = summaryElement.Content.FirstOrDefault() as XmlTextSyntax;
             var text = textElement == null ? string.Empty : XmlCommentHelper.GetText(textElement, true).TrimStart();
 
-            bool startsWithGetOrSet = text.StartsWith(startingTextGetsOrSets, StringComparison.OrdinalIgnoreCase);
+            bool prefixIsGetsOrSets = text.StartsWith(startingTextGetsOrSets, StringComparison.OrdinalIgnoreCase);
+            bool prefixIsGets = text.StartsWith(startingTextGets, StringComparison.OrdinalIgnoreCase) && !prefixIsGetsOrSets;
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/PropertySummaryDocumentationAnalyzer.cs]: 💭 Reversing the condition (put `!prefixIsGetsOrSets` first) will have a small performance benefit.

File: StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/PropertySummaryDocumentationAnalyzer.cs [hunk 2]
```diff
@@ -225,51 +228,85 @@ private static void AnalyzeSummaryElement(SyntaxNodeAnalysisContext context, Xml
...
+                    // Setter exists and is visible. Getter does not exist.
+                    if (!prefixIsSets)
+                    {
+                        ReportSa1623(context, diagnosticLocation, diagnosticProperties, text, expectedStartingText: startingTextSets, unexpectedStartingText1: startingTextGetsOrSets, unexpectedStartingText2: startingTextGets);
+                    }
                 }
             }
         }
+
+        private static void ReportSa1623(SyntaxNodeAnalysisContext context, Location diagnosticLocation, ImmutableDictionary<string, string>.Builder diagnosticProperties, string text, string expectedStartingText, string unexpectedStartingText1, string unexpectedStartingText2 = null)
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/PropertySummaryDocumentationAnalyzer.cs]: I would prefer `ReportSA1623` here, to more clearly preserve the link with the identifier that is reported.

File: StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/PropertySummaryDocumentationAnalyzer.cs [hunk 3]
```diff
@@ -225,51 +228,85 @@ private static void AnalyzeSummaryElement(SyntaxNodeAnalysisContext context, Xml
...
+            }
+            else if ((unexpectedStartingText2 != null) && text.StartsWith(unexpectedStartingText2, StringComparison.OrdinalIgnoreCase))
+            {
+                diagnosticProperties.Add(TextToRemoveKey, text.Substring(0, unexpectedStartingText2.Length));
+            }
+
+            context.ReportDiagnostic(Diagnostic.Create(SA1623Descriptor, diagnosticLocation, diagnosticProperties.ToImmutable(), expectedStartingText));
+        }
+
+        private static void ReportSa1624(SyntaxNodeAnalysisContext context, Location diagnosticLocation, ImmutableDictionary<string, string>.Builder diagnosticProperties, string accessor, string expectedStartingText, string startingTextToRemove)
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/PropertySummaryDocumentationAnalyzer.cs]: I would prefer `ReportSA1624` here, to more clearly preserve the link with the identifier that is reported.

**Comment by bjornhellander** [comment]
Updated according to @vweijsters comments. Also noticed some outdated comments in the code which I removed.

**Pull request merged**